layers/fuzz_layer.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2019 Catena cyber All rights reserved.
+//
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file in the root of the source
+// tree.
+
+package layers
+
+import (
+	"encoding/binary"
+
+	"github.com/google/gopacket"
+)
+
+var fuzzOpts = gopacket.DecodeOptions{
```

Comment:
could this be passed into FuzzLayer?

Reply:
Indeed, this is a good idea.
I chose these options as they seem to maximize the number of panics we can find

Reply:
I'm not completely sure, but I think @gconnell  meant to make `fuzzOpts` an argument of `FuzzLayer`, if possible, so one can change the settings more easily.

Reply:
This is indeed a good idea but not yet possible.
This is called structure ware fuzzing.
Cf https://github.com/golang/go/issues/19109 https://docs.google.com/document/u/1/d/1zXR-TFL3BfnceEAWytV8bnzB2Tfp6EPFinWVJ5V4QC8/pub https://twitter.com/dvyukov/status/1219175271594676224

Reply:
Maybe we can do something with protobuf
Do you want me to try @notti ?

Reply:
> I'm not completely sure, but I think @gconnell meant to make fuzzOpts an argument of FuzzLayer, if possible, so one can change the settings more easily.

I don't think this is possible at the moment. Per https://github.com/dvyukov/go-fuzz/blob/8cb20381268133974b070fff8f404435b3d189c1/go-fuzz-build/main.go#L378 , the type signature must be `[]byte -> int`.